### DISCUSSION TITLE
Linkerd Crashed

### DISCUSSION THREAD

**Discussion by husnerabbi19** [linkerd/linkerd2#8354] [Help]
Hi,
We have deployed Linkerd on a Kubernetes Cluster.

**Node Details**

- OS (CentOS Linux release 7.9.2009)
- Kernel (3.10.0-1160.59.1.el7.x86_64)
- RAM 64GB, CPU 6

**Kubernetes Cluster Details**
- Self managed created by kubeadm
-  version (1.17.5)
- single node cluster

**Linkerd Details**

- version (stable-2.11.1)
- [Deployed by using the Getting Started Guide](https://linkerd.io/2.11/getting-started/)

We have injected Linkerd into our Deployment and Statefulset. 
Initially it works fine, however after 2 days Linkerd components crash.

Here are the Logs of Linkerd POD.

` Warning   Failed    pod/orchestration-controller-5b6d475f94-bxjz9              Error: failed to start container "linkerd-init": Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:275: applying cgroup configuration for process caused \"mkdir /sys/fs/cgroup/memory/kubepods/burstable/pod87f10c9a-b458-4337-a670-1945702f4577/linkerd-init: cannot allocate memory\"": unknown`

![image](https://user-images.githubusercontent.com/46952964/165514329-4acd8ea8-2fc4-4863-bf96-63d742045c23.png)

Let me know, if any other info/logs are required.

**Comment by olix0r** [comment]
Note that the error is not coming from Linkerd, but from your container runtime. `cannot allocate memory` might suggest that the node has no available memory?

**Reply by husnerabbi19** [reply]
Thank you @olix0r  for your reply.

Please find below the 'RAM Usage' details snapshot,
![image](https://user-images.githubusercontent.com/46952964/165704704-051c1fd8-447e-45b0-a83e-056b50294fae.png)

Another thing to note is that if we deploy our application without Linkerd on  the same node and with the same software version, our application works fine without any crash loopback issue.

**Comment by kleimkuhler** [comment]
This is not exactly a solution you may be looking for, but it looks like this could be related to the Node's OS version based off conversations in https://github.com/docker/for-linux/issues/841. CentOS 7.9.2009 seems to be problematic. If you can try testing with a different OS on the node it may be worth seeing if you still run in to any issues.
